### PR TITLE
fix: fix the "finish" event of statistic countdown

### PR DESCRIPTION
**Pull request by hebingchang**
### This is a ...

- [ ] New feature
- [x] Bug fix
- [ ] Site / document update
- [ ] Component style update
- [ ] TypeScript definition update
- [ ] Refactoring
- [ ] Code style optimization
- [ ] Branch merge
- [ ] Other (about what?)

### What's the background?

The `@finish` event in statistic countdown component doesn't work at all.

### Self Check before Merge

- [x] Doc is updated/provided or not needed
- [x] Demo is updated/provided or not needed
- [x] TypeScript definition is updated/provided or not needed
- [x] Changelog is provided or not needed

### CODE REVIEW

File: components/statistic/Countdown.jsx [hunk 1]
```diff
@@ -47,20 +47,21 @@ export default {
     startTimer() {
       if (this.countdownId) return;
       this.countdownId = window.setInterval(() => {
+        const { value } = this.$props;
+        const timestamp = getTime(value);
+
+        if (timestamp < Date.now()) {
+          this.stopTimer();
+        }
         this.$refs.statistic.$forceUpdate();
```
Comment (tangjinzhou) [inline components/statistic/Countdown.jsx]:
```js
startTimer() {
      if (this.countdownId) return;
      this.countdownId = window.setInterval(() => {
        this.$refs.statistic.$forceUpdate();
        this.syncTimer();
      }, REFRESH_INTERVAL);
    },
```

添加 this.syncTimer(); 即可

Comment (hebingchang) [reply components/statistic/Countdown.jsx]: 我的错，已经修改了

File: components/statistic/Countdown.jsx [hunk 2]
```diff
@@ -48,19 +48,15 @@ export default {
       if (this.countdownId) return;
       this.countdownId = window.setInterval(() => {
         this.$refs.statistic.$forceUpdate();
+        this.syncTimer();
       }, REFRESH_INTERVAL);
     },
 
     stopTimer() {
```
Comment (tangjinzhou) [inline components/statistic/Countdown.jsx]: stopTimer 就没必要更改了

Comment (hebingchang) [reply components/statistic/Countdown.jsx]: 确实，已修复，感谢指出

**Pull request merged**